src/OpenTelemetry/Trace/SpanSdk.cs

```diff
@@ -140,15 +140,13 @@ private SpanSdk()
             }
         }
 
-        public SpanContext Context { get; private set; }
+        public override SpanContext Context { get; }
 
         public string Name { get; private set; }
 
         /// <inheritdoc/>
-        public Status Status
+        public override Status Status
```

Comment:
Instead of using override and being forced to add GetStatus(), we could use `new` which would allow us to change the property access modifiers.

For example:
```csharp
public new Status Status
{
    get => this.Status;
    set
    ...
}
```

Reply:
`new` will [hide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/knowing-when-to-use-override-and-new-keywords) the Status from TelemetrySpan and won't override `set`.

exposing getter on `TelemetrySpan` is an option, but it will be one-off case for Status (nothing except `Context` is gettable) and seems like a workaround since we basically don't want API-only users to read span's properties.

Span implementations (`SpanSdk`, `Blank`, etc) are internal and API they expose is not that important.  `SpanData` which is public has a Status getter.

Reply:
You're correct it won't override it, it'll completely replace the property but with get and set accessors. I'm not a fan of using `new` but also not sure removing the getter from Status property and replacing with a `GetStatus`.

@open-telemetry/dotnet-approvers any more opinions on this?

Reply:
@MikeGoldsmith I believe spec requires [set only on Status](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#set-status).  And we are not removing getter - it was never available on `ISpan` (only on the implementation, that was internal). 

We are also keeping getter on publically exposed `SpanData`.

If we add getter on `ISpan`, it will contradict every other property on `ISpan` (`TelemetrySpan`). There are no getters except for the `Context`.

@open-telemetry/dotnet-approvers can you please help us here? 
I'd like to merge it.

Reply:
ISpan doesn't define a getter (and doesn't need to) but we can still have it on TelemetrySpan. I don't see the point in removing it and replacing with a secondary method to retrieve it, it feels over complicated when a simpler approach is to expose the getter.

Reply:
ISpan is becoming TelemetrySpan in this PR.

Do you mean we need Status getter on SpanSdk? It is not possible to add it without introducing it on the `TelemetrySpan`

Reply:
Sorry, yes - my mistake. The above suggestion doesn't work without `ISpan` and that's prompted the `new` suggestion.

Let's just stick with `GetStatus()` in SpanSdk. My preference would be to have the getter in TelemetrySpan but I don't want to block this PR based on a preference.